bundles/org.eclipse.jface/src/org/eclipse/jface/viewers/AbstractTableViewer.java

```diff
@@ -779,6 +810,32 @@ private void internalRemove(final Object[] elements) {
 		}
 	}
 
+	/**
+	 * Returns the data of the last item on the viewer.
+	 *
+	 * @return may return null
+	 */
+	Object getLastElement() {
+		Item lastItem;
+		try {
+			// fast path first
+			int itemCount = doGetItemCount();
+			if (itemCount == 0) {
+				return null;
+			}
+			lastItem = doGetItem(itemCount - 1);
+		} catch (Exception e) {
+			// something went wrong with indexes
+			Item[] items = doGetItems();
```

Comment:
This will force a lazy provider to load all data at once and create an item for each of it.
What is the usecase of "something went wrong with indexes", it does not seem suitable to catch **all** exceptions for handling any indexing issues, maybe it would be more suitable if the expandable node is that special to simply hold a reference in a special field instead of always query the internal items...

Reply:
The "limit" code isn't called for lazy providers